src/sockaddr.c

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright 2019 Xaptum, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "sockaddr.h"
+
+#include <string.h>
+
+int
+enftun_sockaddr_equal(struct sockaddr* a, struct sockaddr* b)
+{
+    if (a->sa_family != b->sa_family)
+        return -1;
+
+    switch (a->sa_family)
+    {
+        case AF_INET:
+        {
+            struct sockaddr_in *ina = (struct sockaddr_in*)a;
+            struct sockaddr_in *inb = (struct sockaddr_in*)b;
+
+            if (ntohl(ina->sin_addr.s_addr) != ntohl(inb->sin_addr.s_addr))
```

Comment:
Is the `ntohl` necessary?  Genuine question, not rhetorical.  Both addresses are on the same machine, so should be the same endianness already.

Reply:
No, I was trying to `memcmp` the `sa_data` buffers of a and b, which did not work, but I had never gone back to using the raw value when I moved to casting them to `sockaddr_in`'s.

Reply:
Gotcha.